server/src/test/java/org/opensearch/common/geo/PointParserTests.java

```diff
@@ -0,0 +1,216 @@
+/*
+ *  Copyright OpenSearch Contributors
+ *  SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.common.geo;
+
+import org.opensearch.OpenSearchParseException;
+import org.opensearch.common.xcontent.XContentBuilder;
+import org.opensearch.common.xcontent.XContentFactory;
+import org.opensearch.common.xcontent.XContentParser;
+import org.opensearch.geometry.Point;
+import org.opensearch.test.OpenSearchTestCase;
+
+import java.io.IOException;
+import java.util.Locale;
+
+public class PointParserTests extends OpenSearchTestCase {
+    private static final PointParser PARSER;
+    private static final PointParser PARSER_WITHOUT_GEOHASH;
+    private static final String FIELD_NAME_X = "x";
+    private static final String FIELD_NAME_Y = "y";
+    private static final String FN_GEOHASH = "geohash";
+    private static final String FN_GEOJSON_TYPE = "type";
+    private static final String FV_GEOJSON_TYPE_POINT = "Point";
+    private static final String FN_GEOJSON_COORDS = "coordinates";
+
+    private static final String FN_IN_ERR_MSG = "test error message";
+
+    static {
+        PARSER = new PointParser(FIELD_NAME_X, FIELD_NAME_Y, true);
+        PARSER_WITHOUT_GEOHASH = new PointParser(FIELD_NAME_X, FIELD_NAME_Y, false);
+    }
+
+    public void testParseFromObjectOfBasicForm() throws IOException {
+        double x = 74.00;
+        double y = 40.71;
```

Comment:
use RandomGeoGenerator class to create randompoints.

Reply:
ACK